tests/serialized_test_data_generation/make_all_datasets.sh

```diff
@@ -1,9 +1,41 @@
 #!/bin/bash
-CONFIGS=configs/c*6*.yml
-make compile_fortran
-for config in $CONFIGS
-do
+
+# This is a utility script which allows to generate serialized data for multiple
+# configuration files matching a certain search pattern.
+
+# Environment variables:
+# CONFIG_PATTERN  search pattern for config files (default: *.yml)
+
+# stop on all errors
+set -e
+
+# directory containing the configuration files
+CONFIG_DIR=configs
+
+# use default pattern if no environment varaible is set
+if [ -z "${CONFIG_PATTERN}" ] ; then
+    CONFIG_PATTERN="*.yml"
+fi
+
+# unset FORTRAN_VERSION, if empty
+if [ -z "${FORTRAN_VERSION}" ] ; then
+    unset FORTRAN_VERSION
+fi
```

Comment:
Why would FORTRAN_VERSION be empty and why do you need to 'unset' it? Does the Makefile interpret FORTRAN_VERSION="" as a valid option and thus doesn't get the default in that case I am guessing?

Reply:
Jenkins does not impose any "filtering" on values provided to FORTRAN_VERSION and - for parametric build - always passes a value even if the user leaves the parameter empty. So I thought it was a good safety measure to assert here.